src/ui/pattern-list-filter.vue

```diff
@@ -12,14 +12,20 @@
 
 	export const DEFAULT_FILTER: Filter = { text: "", cat: "all" };
 
+	function tuneSearchText(state: State, name: string): string {
+		const tune = state.tunes[name];
+		return tune.displayName ? tune.displayName + name : name;
+	}
+
 	export function filterPatternList(state: State, params?: Filter | null): string[] {
 		params = params || DEFAULT_FILTER;
 
 		const ret: string[] = [ ];
 		const tuneNames = getSortedTuneList(state);
 		const text = params && params.text.trim().toLowerCase() || "";
 		for(let i = 0; i < tuneNames.length; i++) {
-			if(text ? (tuneNames[i].toLowerCase().indexOf(text) != -1) : tuneIsInCategory(state.tunes[tuneNames[i]], params.cat))
+			const toSearch = tuneSearchText(state, tuneNames[i]);
+			if(text ? (toSearch.toLowerCase().indexOf(text) != -1) : tuneIsInCategory(state.tunes[tuneNames[i]], params.cat))
```

Comment:
I would prefer `tuneSearchText` to return an array of strings (`[...tune.displayName ? [tune.displayName] : [], name]`) and do `toSearch.some((s) => s.toLowerCase().indexOf(text) != -1)`. I feel like concatenating the two search strings might lead to unexpected search results in some combinations.

Reply:
Yeah good point!

I've combined the "what to search" and the actual searching into a function; I think that makes it easier to read (it took me a while to figure out what was going on...)